### PR TITLE
Updated exception hierarchy

### PR DESCRIPTION
**Pull request by johanste**
Removed library errors, introduced HttpRequestError (which carries a response and status_code). Please note that this is work in progress - tests are not passing. But it should be enough to clean up the hierarchy - I hope...

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: azure-core/azure/core/exceptions.py [hunk 1]
```diff
@@ -73,48 +76,49 @@ def raise_with_traceback(self):
...
-    These errors are safe to retry."""
+class ReadTimeoutError(ServiceRequestError):
+    """The server did not send any data in the allotted amount of time.
+    These errors may not be safe to retry."""
 
 
-class ConnectionReadError(ServiceRequestError):
-    """An error occurred during the request/response.
-    These errors may not be safe to retry."""
+class ServiceResponseError(AzureError):
```
Comment (annatisch) [inline azure-core/azure/core/exceptions.py]: Nothing seems to use ServiceResponseError at the moment? If this is for a request the has reached the service then should it not be of the same family as ReadTimeoutError?

File: azure-core/azure/core/exceptions.py [hunk 2]
```diff
@@ -73,48 +76,49 @@ def raise_with_traceback(self):
             self.__traceback__ = self.exc_traceback
             raise self
 
-
-class AzureLibraryError(AzureError):
```
Comment (annatisch) [inline azure-core/azure/core/exceptions.py]: With the removal of these exceptions - do we no longer intend to define exceptions in azure-core for client-side errors (e.g. a user sending request data that fails to serialize)

Comment (johanste) [reply azure-core/azure/core/exceptions.py]: My suggestion is to use standard exceptions (e.g. ValueError) for these scenarios. But I'd love to get feedback on this proposal...

File: azure-core/azure/core/exceptions.py (hunk 1 above)
Comment (johanste) [reply azure-core/azure/core/exceptions.py]: Yes. Yes, it should.

File: azure-core/azure/core/exceptions.py (hunk 2 above)
Comment (annatisch) [reply azure-core/azure/core/exceptions.py]: Yeah I like that idea... off the top of my head I can't *think* of any scenarios that would fall out side of a TypeError/ValueError - but I'll take a quick scout through the code to see what errors are being raised where any whether any of them may require more thought.

File: azure-core/azure/core/exceptions.py (hunk 1 above)
Comment (annatisch) [reply azure-core/azure/core/exceptions.py]: Okay - so your intention would be to make ReadTimeoutError of type ServiceResponseError?

**Pull request merged**